website/project/model.py

```diff
@@ -2611,12 +2611,15 @@ def add_unregistered_contributor(self, fullname, email, auth,
         self.save()
         return contributor
 
-    def set_privacy(self, permissions, auth=None, log=True, save=True):
-        """Set the permissions for this node.
+    def set_privacy(self, permissions, auth=None, log=True, save=True, skip_mail=False):
+        """Set the permissions for this node. Also, based on skip_mail, queues an email to user about abilities of
+            public projects.
 
         :param permissions: A string, either 'public' or 'private'
         :param auth: All the auth information including user, API key.
         :param bool log: Whether to add a NodeLog for the privacy change.
+        :param bool skip_mail: Whether this should trigger a new_public_project email, used to prevent conferences from
```

Comment:
Another reason not to have this; there might be multiple listeners for the `set_privacy_public` signal that don't have to do with mail at all. This flag would break that code.